course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -0,0 +1,110 @@
+// TODO - this is pretty much a copy of the people controller with some mods.
+// TODO - this still needs a lot of clean up, we don't need everything in here
+
+(function () {
+    var app = angular.module('CourseInfo');
+    app.controller('DetailsController', DetailsController);
+
+    function DetailsController($scope, $routeParams, courseInstances, $compile,
+                               djangoUrl, $http, $q, $log, $uibModal, $sce) {
+
+        dc = this;
+
+        dc.handleAjaxError = function (data, status, headers, config, statusText) {
+            $log.error('Error attempting to ' + config.method + ' ' + config.url +
+                ': ' + status + ' ' + statusText + ': ' + JSON.stringify(data));
+        };
+
+        dc.handleLookupResults = function(results) {
+            var courseResult = results[0];
+            var membersResult = results[1];
+
+            //check if the right data was obtained before storing it
+            if (courseResult.data.course_instance_id == dc.courseInstanceId) {
+                courseInstances.instances[courseResult.data.course_instance_id] = courseResult.data;
+                dc.courseInstance = dc.getFormattedCourseInstance(courseResult.data, membersResult.data)
+            } else {
+                $log.error(' CourseInstance record mismatch for id :'
+                    + dc.courseInstanceId + ',  fetched record for :' + courseResult.data.id);
+            }
+        };
+
+        dc.setCourseInstance = function (id) {
+
+            var course_url = djangoUrl.reverse(
```

Comment:
Most of the time, `courseInstances` should already have the course instance as returned from the list endpoint (it gets populated by the SearchController).  If there's nothing missing from the list serializer that you might need, it would save us an ajax call.

There's also the question of what to do if it *is* missing...eg. if someone just reloaded the lti tool frame.  In the PeopleController, I handled that directly, just like you're doing here.  It's probably worth putting the "fetch if it's missing" logic into the courseInstances directive as a follow-on change once this lands.